mass_editing/models/mass_object.py

```diff
@@ -20,119 +20,116 @@
-        model_obj = self.pool['ir.model']
-        active_model_obj = self.pool.get(model_obj.browse(
-            cr, uid, model_id).model)
-        if active_model_obj._inherits:
-            for key, val in active_model_obj._inherits.items():
-                found_model_ids = model_obj.search(
-                    cr, uid, [('model', '=', key)], context=context)
-                model_ids += found_model_ids
-        return {'value': {'model_ids': [(6, 0, model_ids)]}}
+    @api.onchange('model_id')
+    def onchange_model(self):
+        self.field_ids = [(6, 0, [])]
+        model_list = []
+        if self.model_id:
+            model_obj = self.env['ir.model']
+            model_list = [self.model_id.id]
+            active_model_obj = self.env[self.model_id.model]
+            if active_model_obj._inherits:
+                keys = active_model_obj._inherits.keys()
+                inherits_model_list = model_obj.search([('model', 'in', keys)])
+                model_list.extend((inherits_model_list and
+                                   inherits_model_list.ids or []))
+        self.model_list = model_list
 
-    def create_action(self, cr, uid, ids, context=None):
+    @api.multi
+    def create_action(self):
         vals = {}
-        action_obj = self.pool['ir.actions.act_window']
-        ir_values_obj = self.pool['ir.values']
-        for data in self.browse(cr, uid, ids, context=context):
-            src_obj = data.model_id.model
-            button_name = _('Mass Editing (%s)') % data.name
-            vals['ref_ir_act_window'] = action_obj.create(
-                cr, SUPERUSER_ID,
-                {
-                    'name': button_name,
-                    'type': 'ir.actions.act_window',
-                    'res_model': 'mass.editing.wizard',
-                    'src_model': src_obj,
-                    'view_type': 'form',
-                    'context': "{'mass_editing_object' : %d}" % (data.id),
-                    'view_mode': 'form,tree',
-                    'target': 'new',
-                    'auto_refresh': 1,
-                },
-                context)
-            vals['ref_ir_value'] = ir_values_obj.create(
-                cr, SUPERUSER_ID,
-                {
-                    'name': button_name,
-                    'model': src_obj,
-                    'key2': 'client_action_multi',
-                    'value': (
-                        "ir.actions.act_window," +
-                        str(vals['ref_ir_act_window'])),
-                    'object': True,
-                },
-                context)
-        self.write(
-            cr, uid, ids,
-            {
-                'ref_ir_act_window': vals.get('ref_ir_act_window', False),
-                'ref_ir_value': vals.get('ref_ir_value', False),
-            },
-            context)
+        action_obj = self.env['ir.actions.act_window']
+        src_obj = self.model_id.model
+        button_name = _('Mass Editing (%s)') % self.name
```

Comment:
Looks like we're expecting a singleton record here, but the loop has been removed and there is no `self.ensure_one`